internal/schema/types.go

```diff
@@ -0,0 +1,25 @@
+package schema
+
+import (
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+)
+
+// BaseStringsToStrings converts a slice of basetypes.StringValue to a slice of strings.
+// It extracts the string values from each basetypes.StringValue element.
+func BaseStringsToStrings(stringValues []basetypes.StringValue) []string {
+	result := make([]string, len(stringValues))
+	for i, sv := range stringValues {
+		result[i] = sv.ValueString()
```

Comment:
What if the sv is `null` or `unknown` here? We should do a check before sv.ValueString().